src/group/libp2p_group_gossip_server.erl

```diff
@@ -351,6 +357,62 @@ assign_target(WorkerPid, WorkerRef, TargetAddrs, State=#state{workers=Workers, s
             end
     end.
 
+%% This function will return a single selected seed node from either the
+%% static list or from a DNS lookup. The DNS result _may not_ be selected
+%% because we merge the DNS result with the static list to migitate eclipse
+%% attack attempts.
+maybe_lookup_seed_in_dns(TargetAddrs) ->
+    case application:get_env(libp2p, use_dns_for_seeds, false) of
+        false ->
+            choose_random_element(TargetAddrs);
+        true ->
+            lookup_seed_from_dns(TargetAddrs)
+    end.
+
+choose_random_element([E]) -> E;
+choose_random_element(L) when is_list(L) ->
+    lists:nth(rand:uniform(length(L), L)).
+
+%% We will lookup a seed from a DNS name, and fall back to
+%% picking a random element from the TargetAddrs list
+lookup_seed_from_dns(TargetAddrs) ->
+    case application:get_env(libp2p, seed_dns_cname, undefined) of
+        undefined ->
+            lager:error("Configured to use DNS to lookup seed node IP, but the cname is undefined", []),
+            choose_random_element(TargetAddrs);
+        CName ->
+            case attempt_dns_lookup(CName, ?DNS_RETRIES) of
+                {error, _} = Error ->
+                    lager:error("DNS lookup of ~p resulted in ~p; falling back", [CName, Error]),
+                    choose_random_element(TargetAddrs);
+                {ok, DNSRecord} ->
+                    lager:debug("successful DNS lookup result: ~p", [lager:pr(DNSRecord, inet)]),
+                    %% to help migitate possible eclipse attacks we will blend the DNS result
+                    %% with the static list of seed nodes and randomly pick one of the results
+                    choose_random_element([convert_dns_record(DNSRecord) | TargetAddrs])
```

Comment:
Return the whole list here and let the choose_random_element in assign_target handle it?